deferral/ah-django
# #164859753 Like-Dislike Articles

#### What does this PR do?
- Post likes and dislikes to an article

#### Description of Task to be completed?
 - [X] Allow users to like and dislikes articles

#### How should this be manually tested?
1. Access a clone copy of the repo from [here](https://github.com/andela/ah-django)
```shell
$ git clone https://github.com/andela/ah-django
```
2. Checkout the this feature branch
```shell
$ git checkout ft-article-like-dislikes-164859754
```
3. Create a virtual environment and activate it
```shell
$ pip install virtualenv && virtualenv venv && source venv/bin/activate
```
- Should your pip default to python2, you can create a python3 environent by running
 `virtualenv -p python3 venv` or simply use `pip3` if you have it installed
4. Install the project dependencies
```shell
$ pip install -r requirements.txt
```
5. Use the `.env_sample` to create an `.env` file with environment variables then source the .env.
The `env_sample` details out the required environment vairables. Copy them and paste in an `env` file. Export the environment variables with the command:
```
$ source .env
```

6. Make database migrations and start the django server
```shell
$ python3 manage.py makemigrations
$ python3 manage.py migrate
$ python3 manage.py runserver
```

7. Testing the endpoint
- First create an article by sending a `POST` request to the endpoint `127.0.0.1:8000/api/articles/`
```shell
 'article': {
                'title': 'Funny thing on the keyboard',
                'description': 'Suggestions for the name of that funny thing',
                'body': 'Now how do we go about this thing...'
            }
        }

```
- Copy the `slug` value from the return data and use it in the like/dislike urls in the format shown below
`api/articles/<slug>/like`
- Post a like or dislike to this article by sending a post request to the endpoints

#### like
 `127.0.0.1:8000/api/articles/<enter_slug>/like/`

#### dislike
 `127.0.0.1:8000/api/articles/<enter_slug>/dislike/`

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#164859755](https://www.pivotaltracker.com/story/show/164859755)

#### Screenshot

![llike_article](https://user-images.githubusercontent.com/44059971/55589534-23fc4b00-5739-11e9-887e-dfe7ad2bfb04.png)
